libs/react-sdk/src/use-client.ts

```diff
@@ -17,15 +31,34 @@ export function useClient({
         // eslint-disable-next-line react-hooks/exhaustive-deps
         [filteredAudiences.join()],
     )
-    const [, forceRender] = useState(0)
+
     useEffect(() => {
-        return browserClient.subscribeToInitialisedChanged((init: boolean) =>
-            forceRender((curr) => curr + 1),
+        return browserClient.subscribeToInitialisedChanged(
```

Comment:
There is a subtle race condition here I think.

The React component can render, then `useEffect` runs on the event loop later, if `initialised` changes between those two things, you will miss the `initialised` event firing.

We need to subscribe, and also ensure we haven't missed a state changed with `if (browserClient.initialised !== initialised) {` then set initialising for that scenario.